File: PassEnCDeC.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

namespace MilkTeaOrderAndInventorySystem
{

    public class PassEnCDeC
    {
        private readonly TripleDESCryptoServiceProvider _des = new TripleDESCryptoServiceProvider();
        private readonly UnicodeEncoding _uni = new UnicodeEncoding();
        private readonly byte[] _key;
        private readonly byte[] _iv;

        public PassEnCDeC(byte[] key, byte[] iv)
        {
            _key = key;
            _iv = iv;
        }

        public byte[] Encrypt(byte[] input)
        {
            return CryptoTransform(input, _des.CreateEncryptor(_key, _iv));
        }

        public string Encrypt(string text)
        {
            byte[] input = _uni.GetBytes(text);
            byte[] output = CryptoTransform(input, _des.CreateEncryptor(_key, _iv));
            return Convert.ToBase64String(output);
        }

        public byte[] Decrypt(byte[] input)
        {
            return CryptoTransform(input, _des.CreateDecryptor(_key, _iv));
        }

        public string Decrypt(string text)
        {
            byte[] input = Convert.FromBase64String(text);
            byte[] output = CryptoTransform(input, _des.CreateDecryptor(_key, _iv));
            return _uni.GetString(output);
        }

        private byte[] CryptoTransform(byte[] input, ICryptoTransform transform)
        {
            byte[] result;

            using (var ms = new MemoryStream())
            {
                using (var cs = new CryptoStream(ms, transform, CryptoStreamMode.Write))
                {
                    cs.Write(input, 0, input.Length);
                    cs.FlushFinalBlock();
                    ms.Position = 0L;
                    result = ms.ToArray();
                }
            }

            return result;
        }
    }
}

File: PassEnCDeC.vb
Imports System.IO
Imports System.Security.Cryptography
Imports System.Text

Public Class PassEnCDeC
    Private ReadOnly _des As New TripleDESCryptoServiceProvider
    Private ReadOnly _uni As New UnicodeEncoding
    Private ReadOnly _key() As Byte
    Private ReadOnly _iv() As Byte

    Public Sub New(key() As Byte, iv() As Byte)
        _key = key
        _iv = iv
    End Sub

    Public Function Encrypt(input() As Byte) As Byte()
        Return CryptoTransform(input, _des.CreateEncryptor(_key, _iv))
    End Function

    Public Function Encrypt(text As String) As String
        Dim input() = _uni.GetBytes(text)
        Dim output() = CryptoTransform(input, _des.CreateEncryptor(_key, _iv))
        Return Convert.ToBase64String(output)
    End Function

    Public Function Decrypt(input() As Byte) As Byte()
        Return CryptoTransform(input, _des.CreateDecryptor(_key, _iv))
    End Function

    Public Function Decrypt(text As String) As String
        Dim input() = Convert.FromBase64String(text)
        Dim output() = CryptoTransform(input, _des.CreateDecryptor(_key, _iv))
        Return _uni.GetString(output)
    End Function

    Private Function CryptoTransform(input() As Byte, transform As ICryptoTransform) As Byte()
        Dim result As Byte()

        Using ms As New MemoryStream
            Using cs As New CryptoStream(ms, transform, CryptoStreamMode.Write)
                cs.Write(input, 0, input.Length)
                cs.FlushFinalBlock()
                ms.Position = 0
                result = ms.ToArray()
            End Using
        End Using

        Return result
    End Function
End Class
